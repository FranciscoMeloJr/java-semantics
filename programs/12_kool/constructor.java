// Testing new, constructor, method invocation.

class C {
  int i, j;

  void C(int x) {
    i = x;
    j = ++x;
  }

  void add(int d) {
    i = i+d;
    j = j-d;
  }

  void print2() {
    System.out.println(i+ " "+ j);
  }
}

class main {
  int a, b;
  C o;

  main(String[] args){
    a = b = 5;
    o = new C(a);
    o.print2();
    o.add(++b);
    o.print2();
    System.out.println("Done!");
  }
}

public class constructor {
  public static void main(String[] args) {
    new main(args);
  }
}

// 5 6
// 11 0
// Done!
