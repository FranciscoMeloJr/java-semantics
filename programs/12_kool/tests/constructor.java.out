5 6
11 0
Done!
